include/filewriter/filewriter_impl.hpp

```diff
@@ -18,15 +18,16 @@ FileWriter::FileWriter(std::string logdir,
   std::string currentTime =
       std::to_string(std::chrono::duration_cast<std::chrono::seconds>(
       p1.time_since_epoch()).count());
-  this->logdir = logdir + "/events.out.tfevents." + currentTime + ".v2";
+  this->logdir = logdir;
+  this->filename = this->logdir + "/events.out.tfevents." + currentTime + ".v2";
   close_ = true;
```

Comment:
What's your intent/plan for this?

Reply:
Just differentiating between directory and file 

Log directory is the directory name where files would be stored
And file is filename 


May be in future , they could be concrete difference and may come in handy
As of now both are same